src/Stratis.Bitcoin/Builder/FullNodeBuilder.cs

```diff
@@ -203,7 +204,22 @@ private IServiceCollection BuildServices()
 
             // configure features startup
             foreach (var featureRegistration in this.Features.FeatureRegistrations)
+            {
+                try
+                {
+                    featureRegistration.EnsureDependencies(this.Features.FeatureRegistrations);
+                }
+                catch (MissingDependencyException e)
+                {
+                    var logger = this.NodeSettings.LoggerFactory.CreateLogger(typeof(FullNodeBuilder).FullName);
+                    logger.LogCritical("Feature {0} cannot be configured because it depends on other features that were not registered: {1}",
+                        featureRegistration.FeatureType.Name, e.Message);
+
+                    continue;
```

Comment:
`continue;` will just continue to execution of the code no? don't we want to throw at this point?

Reply:
Yes. The idea behind it was to run node without features that have broken dependencies. But I guess you right, so I'll change it to throw forward caught exception.